py/torch_tensorrt/dynamo/lowering/_decompositions.py

```diff
@@ -162,6 +164,47 @@ def var_decomposition(
     return variance
 
 
+@register_torch_trt_decomposition(
+    torch.ops.aten.slice_scatter.default, registry=TORCH_TRT_DECOMPOSITIONS
+)
+def slice_scatter_decomposition(
+    input_tensor: torch.Tensor,
+    src_tensor: torch.Tensor,
+    dim: int,
+    start: Optional[int],
+    end: Optional[int],
+    step: int,
+):
+    dim_size = input_tensor.shape[dim]
+    start = get_positive_dim(start, input_tensor.shape)
+    end = get_positive_dim(end, input_tensor.shape)
+
+    src_dim = src_tensor.shape
+    step_dim = (end - start) // step
+    end_dim = end
+    if step_dim > src_dim[dim]:
+        end_dim = src_dim[dim]
+    else:
+        indices = torch.arange(0, step_dim)
+        indices = indices.to(torch.int32)
```

Comment:
If the indices are `int64`, it is fine to leave them as-is and not change the data type, since later operators may expect or require `int64`

Reply:
Does `torch` expect `int64` for the indices in `index_select`, or TensorRT? If it is TensorRT, then there is no need to perform the cast, because the outputs of the above operation will already have been handled in the `TRTInterpreter`

Reply:
Yes torch would expect int64 input for the indices. Since this is a constant, I think TRTInterpretor should be able to handle it. Yes I will remove this.